src/content/docs/en/reference/dev-overlay-reference.mdx

```diff
@@ -0,0 +1,260 @@
+
+```ts
+export default {
+	id: 'super-plugin',
+	name: 'My Super Plugin',
+	icon: '<svg>...</svg>',
+	init(canvas) {
+    canvas.appendChild(document.createTextNode('Hello World!'))
+  }
+}
+```
+
+#### `eventTarget`
+
+An [`EventTarget`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget) that can be used to send and receive events from the dev overlay.
+
+## Client-side Events
+
+Using the `eventTarget` argument on the `init` hook, plugins can send and receive events from the dev overlay. The following events are available:
+
+### `plugin-toggle`
+
+This event is fired when the user clicks on the plugin icon in the UI. The new state of the plugin is included in the event in the `detail` property.
+
+```ts title="src/my-plugin.js" {3-6}
+  export default {
+    // ...
+    init(canvas, eventTarget) {
+      eventTarget.addEventListener('plugin-toggle', (event) => {
+        if (event.detail.state === true) {
+          console.log("The plugin is now enabled!")
+        }
+      })
+    }
+  }
+```
+
+### `plugin-notification`
+
+This event can be sent to inform the user that the plugin requires attention.
+
+```ts title="src/my-plugin.js" {3-6}
+  export default {
+    // ...
+    init(canvas, eventTarget) {
+      eventTarget.dispatchEvent(
+          new CustomEvent('plugin-notification', {
+            detail: {
+              state: true,
+            },
+          })
+        );
+    }
+  }
+```
+
+`state` can be either `true` or `false`. When `true`, the plugin icon will be highlighted using a red dot. Conversely, when `false`, the highlight will be removed.
+
+## Client-Server Communication
+
+Using [Vite's methods for client-server communication](https://vitejs.dev/guide/api-plugin.html#client-server-communication), dev overlay plugins can communicate with the server.
+
+In addition to being able to send and receive arbritary messages, the dev overlay also sends the following messages, where `PLUGIN_ID` is the ID of a plugin:
+
+### `astro-dev-overlay:PLUGIN_ID:init`
+
+This message is sent when the plugin is initialized. The data for this message is empty.
+
+```ts title="integration.ts"
+
+```
+
+### `astro-dev-overlay:PLUGIN_ID:toggle`
+
+This message is sent when the user clicks on the plugin icon in the UI. The data for this message is a boolean indicating whether the plugin is enabled or not.
+
+:::note
+The built-in `connection` event from Vite fire *before* dev overlay plugins are initialized. As such, trying to receive events from this event will not work inside plugins. Instead, use the `astro-dev-overlay:PLUGIN_ID:init` event.
+:::
+
+## UI Toolkit
+
+The dev overlay includes a set of webcomponents that can be used to build plugins with a consistent look and feel.
+
+### `astro-dev-overlay-window`
+
+Shows a window with optionally a title and an icon. `windowIcon` can either be a string of a SVG file or an icon from [the icon list](/en/reference/dev-overlay-plugin-api/#icons).
```

Comment:
```suggestion
Shows a window with optionally a title and an icon.

`windowTitle` is a string that will be shown at the top of the overlay window. `windowIcon` can either be a string of a SVG file or an icon from [the icon list](#icons).
```
I agree this maybe feels stupidly unnecessary, but it is a part of the API and it should be mentioned explicitly somewhere.